addons/storyshots/src/index.js

```diff
@@ -21,6 +27,16 @@ const hasDependency = name =>
   (pkg.devDependencies && pkg.devDependencies[name]) ||
   (pkg.dependencies && pkg.dependencies[name]);
 
+function patchStoriesOfFunction(storybookImpl) {
```

Comment:
Indeed it's a better way.. I just didn't want to change the core only for the addon needs.. But maybe in this case it's a good solution.